rest/route/pod.go

```diff
@@ -80,3 +88,94 @@ func (h *podAgentCedarConfig) Run(ctx context.Context) gimlet.Responder {
 	}
 	return gimlet.NewJSONResponse(data)
 }
+
+////////////////////////////////////////////////
+//
+// POST /rest/v2/pods
+
+type podPostHandler struct {
+	body []byte
+	sc   data.Connector
+	p    model.APICreatePod
+}
+
+func makePostPod(sc data.Connector) gimlet.RouteHandler {
+	return &podPostHandler{
+		sc: sc,
+		p:  model.APICreatePod{},
+	}
+}
+
+func (h *podPostHandler) Factory() gimlet.RouteHandler {
+	return &podPostHandler{
+		sc: h.sc,
+		p:  model.APICreatePod{},
+	}
+}
+
+// Parse fetches the podID and JSON payload from the HTTP request.
+func (h *podPostHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+
+	if err := json.Unmarshal(b, &h.p); err != nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("API error while unmarshalling JSON"),
+		}
+	}
+
+	if utility.FromStringPtr(h.p.Image) == "" {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintln("Invalid API input: missing or empty image input"),
+		}
+	} else if utility.FromStringPtr(h.p.Platform) == "" {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintln("Invalid API input: missing or empty platform"),
+		}
+	} else if utility.FromIntPtr(h.p.CPU) <= 0 {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintln("Invalid API input: missing or invalid CPU"),
+		}
+	} else if utility.FromIntPtr(h.p.Memory) <= 0 {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintln("Invalid API input: missing or invalid memory"),
+		}
+	}
+
+	for _, envVar := range h.p.EnvVars {
+		if utility.FromStringPtr(envVar.Name) == "" {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusBadRequest,
+				Message:    fmt.Sprintf("Invalid API input: missing or empty environment variable name"),
+			}
+		}
+	}
+
+	return nil
+}
+
+// Run creates a new resource based on the Request-URI and JSON payload.
+func (h *podPostHandler) Run(ctx context.Context) gimlet.Responder {
+	id, err := h.sc.CreatePod(h.p)
```

Comment:
nit: can you call the variable something else like `res`, `resp`, or `createResp` since it might not just be an ID in the future?